app/server/api/contract_api.py

```diff
@@ -0,0 +1,118 @@
+from flask import Blueprint, request, make_response, jsonify, current_app
+from flask.views import MethodView
+
+from server import db, bt
+from server.utils.auth import requires_auth
+from server.models.token import Token
+from server.models.exchange import ExchangeContract
+from server.schemas import token_schema
+
+deploy_contracts_blueprint = Blueprint('contracts', __name__)
+
+
+class ContractsAPI(MethodView):
+
+    @requires_auth(allowed_roles={'ADMIN': 'sempoadmin'})
+    def post(self):
+        # post_data = request.get_json()
+
+        name = 'CIC Test 2'
```

Comment:
Placeholder data?

Reply:
Oh good catch. all of these need to be swapped to pull from POST request.

Didn't get picked up during writing tests because the tests only test the use of the exchange once it's deployed, rather than the creation of the exchange itself.

Also, deploying a token contract is done automatically when u post a token. for consistency i'm thinking i should either:
- delete the contract_api, and move this to a POST on /api/exchange/contract instead. 
- make it so that POSTING to /api/token/ only registers an existing token, which right now is achieved by providing an address when posting (we definitely need this functionality, as a lot of the tokens we use will be external ones like DAI), and moving all contract deployment to /api/contract/token, /api/contract/exchange etc etc....

Thoughts?

Reply:
update: took option 2! saves having token registration/creation squeezed into one endpoint